### PR TITLE
remove raw color codes in shop cost (Commons-Box issue in latest commit)

### PR DESCRIPTION
**Pull request by HSGamer**
`legacyAmpersand()`'s `getLore` will serialize the lore with the color codes.
Therefore, the return lore will be something like `&650 orbs` and will then be reduced to `650`.

### CODE REVIEW

File: src/main/java/plugily/projects/villagedefense/arena/managers/ShopManager.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ private void registerShop() {
       if(itemStack.hasItemMeta() && itemStack.getItemMeta().hasLore()) {
         for(String s : ComplementAccessor.getComplement().getLore(itemStack.getItemMeta())) {
           if(s.contains(plugin.getChatManager().colorMessage(Messages.SHOP_MESSAGES_CURRENCY_IN_SHOP)) || s.contains("orbs")) {
-            costString = ChatColor.stripColor(s).replaceAll("[^0-9]", "");
+            costString = ChatColor.stripColor(s).replaceAll("&[0-9a-zA-Z]", "").replaceAll("[^0-9]", "");
```
Comment (montlikadani) [inline src/main/java/plugily/projects/villagedefense/arena/managers/ShopManager.java:139]: `ChatColor#stripColor` already do this.

Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/managers/ShopManager.java:139]: @montlikadani
In Spigot and Paper prior to the latest build, yes.
But your latest commit on Commons_Box changed the serializer to `LegacyComponentSerializer.legacyAmpersand()`.
The result is that the string from `getComplement().getLore()` will return the raw string (`&650 orbs`), and `ChatColor#stripColor` itself will not remove the `&6`

Comment (HSGamer) [reply src/main/java/plugily/projects/villagedefense/arena/managers/ShopManager.java:139]: You can test yourself by 
- Using the latest Paper 1.16.5 with Adventure
- Compiling with the latest `Commons-Box`
- Checking the `s` in the lore loop and the parsed `cost`

**Comment by HSGamer** [comment]
For more details, this is the result, using the latest `Paper 1.16.5` and `Commons-Box`
Screenshot_20210503_174612

The code
Screenshot_20210503_174640

**Pull request merged**